src/lib/dvf/createFastWithdrawalPayload.js

```diff
@@ -0,0 +1,120 @@
+const FP = require('lodash/fp')
+const {
+  fromQuantizedToBaseUnitsBN,
+  Joi,
+  starkTransferTxToMessageHash,
+  toQuantizedAmountBN
+} = require('dvf-utils')
+const swJs = require('starkware_crypto')
+
+const calculateFact = require('../stark/calculateFact')
+const validateWithJoi = require('../validators/validateWithJoi')
+
+const DVFError = require('./DVFError')
+
+// TODO
+const calculateFee = amount => 0
+
+const address0 = '0x'.padEnd(42, '0')
+
+const getValidTokenInfo = dvf => token => {
+  const tokenInfo = dvf.token.getTokenInfoOrThrow(token)
+
+  if (!tokenInfo.starkVaultId) {
+    throw new DVFError(
+      'NO_STARK_VAULT_ID_FOR_TOKEN',
+      { token, context: 'fastWithdrawal' }
+    )
+  }
+
+  if (!tokenInfo.deversifiStarkVaultId) {
+    throw new DVFError(
+      'NO_DEVERSIFI_STARK_VAULT_ID_FOR_TOKEN',
+      { token, context: 'fastWithdrawal' }
+    )
+  }
+
+  return tokenInfo
+}
+
+const schema = Joi.object({
+  amount: Joi.amount().required(),
+  // NOTE: we are not specifying allowed tokens here since these can change
+  // dynamically. However a call to `getTokenInfoOrThrow` will ensure that
+  // the token in valid.
+  token: Joi.string().required()
+})
+
+const errorProps = { context: 'fastWithdrawal' }
+const validateArg0 = validateWithJoi
+  (schema)
+  ('INVALID_METHOD_ARGUMENT')
+  ({ ...errorProps, argIdx: 0 })
+
+module.exports = async (dvf, withdrawalData, starkPrivateKey) => {
+  // Not asserting a specific type since it could be either a string or BigInt
+  // depending on the used sw crypto implementation.
+  if (!starkPrivateKey) {
+    throw new DVFError(
+      'STARK_PRIVATE_KEY_IS_REQUIRED', { ...errorProps, argIdx: 1 }
+    )
+  }
+
+  const { amount, token } = validateArg0(withdrawalData)
+
+  const tokenInfo = getValidTokenInfo(dvf)(token)
+
+  const { starkPublicKey, starkKeyPair } = await dvf.stark.createKeyPair(
+    starkPrivateKey
+  )
+
+  const transactionFee = toQuantizedAmountBN(tokenInfo, calculateFee(amount))
+
+  // This should be in hours
+  const expirationTimestamp =
+    Math.ceil(Date.now() / (1000 * 3600)) +
+    parseInt(dvf.config.defaultStarkExpiry)
+
+  const usersEthAddress = dvf.config.ethAddress
```

Comment:
The user can request a fast withdrawal to any address, not just their own. Probably the best pattern would be to have an optional parameter for recipient address, but default to dvf.config.ethAddress if it is not supplied

Reply:
this should be easy to add, I'd like to do some e2e tests first before 

btw. is the address supposed to be checksummed or not? I believe currently all eth addresses stored the db are lower case

Reply:
I believe for what the user needs to sign, and the fact on the blockchain, all addresses should be represented and hashed as check-summed (since that is how the prover and blockchain will represent them).

You can use `web3.utils.toChecksumAddress(address)` to transform them

Reply:
I believe the use of `encodePacked` in 

https://github.com/DeversiFi/dvf-client-js/blob/992a2fcc37dad839a1aacb559b3ef288de0af59d/src/lib/stark/calculateFact.js#L3-L8

ensures that the fact is calculated the same way as on chain